src/microsoft_mcp/email_framework/test_runner.py

```
@@ -0,0 +1,308 @@
+        print("Testing Performance...")
+        
+        # Test with practice report (most complex template)
+        template = PracticeReportTemplate()
+        data = template.generate_sample_data()
+        
+        # Warm up
+        self.renderer.render("practice_report", data)
+        
+        # Measure rendering time
+        times = []
+        for i in range(10):
```

Ruff (0.12.2)

Loop control variable `i` not used within loop body

Rename unused `i` to `_i`

(B007)